src/compiler/checker.ts

```diff
@@ -21978,7 +21978,16 @@ namespace ts {
                         const type = getTypeOfDottedName((<PropertyAccessExpression>node).expression, diagnostic);
                         if (type) {
                             const name = (<PropertyAccessExpression>node).name;
-                            const prop = getPropertyOfType(type, isPrivateIdentifier(name) ? getSymbolNameForPrivateIdentifier(type.symbol, name.escapedText) : name.escapedText);
+                            let prop: Symbol | undefined;
+                            if (isPrivateIdentifier(name)) {
+                                if (!type.symbol) {

```

Comment:
Hm, I dislike relying on `type.symbol`, but it seems like we can only guarantee `getNodeLinks((<PropertyAccessExpression>node).expression).resolvedSymbol` exists for `Identifier` LHSs, since as `getTypeOfDottedName` recurs, it doesn't set it on other interesting LHS node kinds, like `checkPropertyAccessExpressionOrQualifiedName` sometimes does.

Reply:
I initially had an `isAnyLike` check since I discovered the issue on  `any`, but `never` and `unknown` also trigger the issue.